core/src/main/java/io/questdb/std/Chars.java

```diff
@@ -490,6 +490,34 @@ public static char toLowerCaseAscii(char character) {
         return character > 64 && character < 91 ? (char) (character + 32) : character;
     }
 
+    public static void toUpperCase(@Nullable final CharSequence str, final CharSink sink) {
+        if (str == null || str.length() == 0) {
+            return;
+        }
+
+        for (int i = 0; i < str.length(); i++) {
+            sink.put(toUpperCase(str.charAt(i)));
+        }
+    }
+
+    public static char toUpperCase(final char character) {
+        return Character.toUpperCase(character);
+    }
+
+    public static void toLowerCase(@Nullable final CharSequence str, final CharSink sink) {
+        if (str == null || str.length() == 0) {
```

Comment:
this implementation can be improved in terms of hot path isolation:

- string is not null and longer than zero (hot)
- string is null or empty (less frequent - cold)

hot path might look like:

```java
if (str != null) {
    for (int i = 0, n=str.length(); i < n; i++) {
                sink.put(toLowerCase(str.charAt(i)));
    }
}
```

note that is `str.length()` is not cached into local variable, java will be calling `length()` method for every iteration

Reply:
Since we are reducing method calls, I'll also remove the methods toUpperCase(char) and toLowerCase(char) and call the respective Character methods directly.